### ISSUE TITLE
Search in Arrays

### ISSUE THREAD

**Issue by DominicBoettger** [techfort/LokiJS#91]
I have a array in my documents:

///
{
  title: 'test',
  topics: [{id: 1, title: 'topic1'}, {id:3, title: ' topic3'}]
}
///

Is it possible to search for documents where topics.id = 3.
In mongodb i can do this with 'topics.id': 3. Is it possible to search in arrays like this with loki?

Thanks
Best regards
Dominic

**Comment by obeliskos** [comment]
yes we support dot notation in find()

```javascript
collection.find({'topics.id':3});
```

**Comment by obeliskos** [comment]
Sorry... I should have read more carefully.

Currently 'no',  our dot notation currently does not work on arrays.  This seems to be something we could add without too much effort so I will look into adding it over the next few days or so.

Until then you could use 'where' clauses which use javascript filter functions to do this filtering.

**Comment by obeliskos** [comment]
I did initial checkin of changes which support your example.  In doing so I noticed dot notation may not be detecting properly when you include an operator.  I'll work on that next but in meantime, form your query with implicit equals operator like : 

```javascript
coll.find({'topics.id', 3});
```

If you prefer to wait I will probably do further testing and fix operator detection over the next few days.
